include/proxy-wasm/wasm_vm.h

```diff
@@ -143,12 +143,25 @@ enum class AbiVersion { ProxyWasm_0_1_0, ProxyWasm_0_2_0, ProxyWasm_0_2_1, Unkno
 
 class NullPlugin;
 
+enum class FailState : int {
+  Ok = 0,
+  UnableToCreateVm = 1,
+  UnableToCloneVm = 2,
+  MissingFunction = 3,
+  UnableToInitializeCode = 4,
+  StartFailed = 5,
+  ConfigureFailed = 6,
+  RuntimeError = 7,
+};
+
 // Integrator specific WasmVm operations.
 struct WasmVmIntegration {
   virtual ~WasmVmIntegration() {}
   virtual WasmVmIntegration *clone() = 0;
   virtual proxy_wasm::LogLevel getLogLevel() = 0;
   virtual void error(std::string_view message) = 0;
+  // Allow integrations to handle specific FailStates differently.
+  virtual void error(FailState fail_state, std::string_view message) { error(message); }
```

Comment:
This is causing `error: unused parameter 'fail_state' [-Werror,-Wunused-parameter]`